### PR TITLE
Makefile: simplify and compile binary in one batch

### PR DESCRIPTION
**Pull request by GetPastTheMonkey**
Instead of compiling the different source files into object files
and then linking those, we switch to compiling the binaries in a
single go. This fixes some dependency tracking issues. Speed is not
an issue since the files are relatively small.

Additional changes:
 * We also split out the build configs (path and CLAGS) into a
   default_config.inc and user_config.inc file that are included
   from the main Makefile.
   The `user_config.inc` file serves as documentation as all lines
   are commented out. It's still part of .gitignore, so that changes
   aren't picked up by default.

 * Only one i2c type is built from the `all` target (depending on
   `CONFIG_I2C_TYPE`).

 * The user has the option of providing a custom `user_config.inc`
   to further customize the build.

 * Updated circle CI script to build both the empty stubs and the
   Linux user space sample implementation for both hw_i2c and
   sw_i2c.

 * Move the defined constants `SHT3X_MEASUREMENT_DURATION_USEC` and
   `SHTC1_MEASUREMENT_DURATION_USEC` to their respective header file.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,18 +15,33 @@ jobs:
...
+              cd -
       - run:
-          name: Move zip files to artifacts directory
+          name: build Linux release drivers
           command: |
-            mkdir -p /releases
-            mv release/*.zip /releases/
-
-      - store_artifacts:
-          path: /releases
```
Comment (rnestler) [inline .circleci/config.yml]: Any reason you removed the artifact store step?

File: .gitignore [hunk 2]
```diff
@@ -2,6 +2,8 @@
 /release
 /sht3x/sht3x_example_usage_hw_i2c
 /sht3x/sht3x_example_usage_sw_i2c
+/sht3x/user_config.inc
 /shtc1/shtc1_example_usage_hw_i2c
 /shtc1/shtc1_example_usage_sw_i2c
+/shtc1/user_config.inc
```
Comment (rnestler) [inline .gitignore]: We ignore these files, but they get added by this commit anyways. Is this intended?

Comment (abrauchli) [reply .gitignore]: yes, we want the file checked in as template, but typically ignore changes the user makes. We'll have to `git add -f` to update the template

File: .circleci/config.yml (hunk 1 above)
Comment (GetPastTheMonkey) [reply .circleci/config.yml]: I wrote the config similar to the one we have in the embedded-sts driver:
https://github.com/Sensirion/embedded-sts/blob/3421a78ff86e4163b5d78b8cfc077142e7079498/.circleci/config.yml

Comment (abrauchli) [reply .circleci/config.yml]: I guess we could keep them on "tags" builds and then upload those to github - that might even be the saner idea over creating it locally - keeping release artifacts of non-release builds the rest is a bit of space-waste.. although a limited one if we restrict the number of builds to keep.

Comment (rnestler) [reply .circleci/config.yml]: @GetPastTheMonkey Could you add a release job which only triggers on tags? Take a look at https://circleci.com/docs/2.0/configuration-reference/#tags. Just with a `make release` and `store_artifacts`.

Comment (GetPastTheMonkey) [reply .circleci/config.yml]: It should be working now

File: .circleci/config.yml [hunk 3]
```diff
@@ -15,18 +15,64 @@ jobs:
...
+    docker:
+      - image: gcc:5.5
+    steps:
+      - checkout
+      - run:
+          name: update common repo
+          command: git submodule update --init
+      - run: apt update
+      - run: apt install -y zip
+      - run: make release
```
Comment (rnestler) [inline .circleci/config.yml]: `Move zip files to artifacts directory` is missing (so the artifacts are empty).

Comment (GetPastTheMonkey) [reply .circleci/config.yml]: Done

File: .gitignore [hunk 4]
```diff
@@ -2,6 +2,8 @@
 /release
 /sht3x/sht3x_example_usage_hw_i2c
 /sht3x/sht3x_example_usage_sw_i2c
+/sht3x/user_config.inc
```
Comment (abrauchli) [inline .gitignore]: I think the suffix `_hw_i2c` and `_sw_i2c` is gone

Comment (GetPastTheMonkey) [reply .gitignore]: Good catch!

**Pull request merged**